### PR TITLE
Add tests and docs for installer internal release helpers

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Adds tests and documentation for internal installer release-helper binaries and their workflow contract
- Introduces Python-based tests to validate internal Cargo metadata helpers used by workflows
- Updates an example in behaviour docs to reflect a real prebuilt path for consistency

### Documentation
- docs/developers-guide.md
  - Adds a new section: "Installer release helper binaries" explaining why autobins = false is required, lists current explicit binary targets, and outlines the purpose and validation coverage of workflow contract tests.
  - Provides guidance on how to extend the contract when adding new packaging binaries, including a worked example and alignment steps for related workflows.
  - Emphasizes keeping workflow YAML, installer/Cargo.toml, and metadata-based tests in lock-step when modifying release helpers.

### Tests
- installer/tests/behaviour_docs.rs
  - Updates the pre-built path example from a generic placeholder to a concrete path (path = "/whitaker/lints/") to improve determinism in tests.

- tests/workflows/test_workflow_test_helpers.py (new)
  - Introduces unit tests for internal cargo metadata helpers:
    - TestCargoMetadata: checks that cargo must be on PATH, surfaces cargo metadata failures, rejects non-mapping JSON, and returns parsed workspace metadata.
    - TestWorkspacePackageMetadata: validates presence of a packages list, reports missing workspace packages, and returns the matching package while ignoring malformed entries.
  - These tests exercise internal APIs used by workflow contract tests to guard against drift in metadata resolution.

- tests/workflows/workflow_test_helpers.py
  - Minor documentation change to test comments: references issue #101 for the PATH-resolved tooling note to reflect current tracking.

## Why this matters
- The installer exposes internal release-helper binaries that are used by GitHub workflows and packaging scripts. Ensuring explicit target declarations and contract-driven tests helps prevent breakages when Cargo or tooling defaults change.
- The new Python tests provide clearer, deterministic validation of how workflow tests resolve workspace metadata, reducing flakiness and surfacing errors with actionable messages.

## Test plan
- Run Rust tests: cargo test (in the installer crate) to ensure the new behaviour_docs change remains valid.
- Run Python tests: pytest to exercise tests/workflows/test_workflow_test_helpers.py and related workflow test helpers.
- Verify that the contract tests for installer packaging bins still pass and that the explicit bin targets are correctly reflected in workspace metadata.
- If modifying release helpers in the future, update the docs and tests in lock-step as described in the new documentation.

## Notes
- No user-facing API changes; this work strengthens internal APIs and the associated contract tests and docs to reduce drift and surprises in workflows and packaging.

◳ Generated by [DevBoxer](https://www.devboxer.com) ◰

---
ℹ️ Tag @devboxerhub to ask questions and address PR feedback

📎 **Task**: https://www.devboxer.com/task/432a36f0-45d3-4fd8-a2b7-27de0389b0ef

## Summary by Sourcery

Document and test the installer’s internal release-helper binaries and the workflow metadata helpers they depend on.

Enhancements:
- Clarify workflow helper comments to reference the tracking issue for PATH-resolved tooling.

Documentation:
- Describe the installer release-helper binaries, why explicit bin targets with `autobins = false` are required, and how to extend the workflow contract when adding new packaging binaries.

Tests:
- Add Python unit tests for the workflow cargo metadata helpers to validate error handling and workspace package resolution, and adjust an installer behaviour doc test to use a concrete pre-built path example.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response. Ensure that this is validated against the *current* version of the codegraph.

If further refinement to address this finding would be deleterious, please supply a clear explanatory one to two paragraph markdown message I can paste into the CodeScene web ui's diagnostic suppression function so this diagnostic can be silenced.

[tests/workflows/test_workflow_test_helpers.py](https://github.com/leynos/whitaker/pull/198/files/132d3402e00050a38f69e0f6e3cfff450aed29e7#diff-1c1294f90a1194e8871d4f1653c5c7dd3ee97d3b737e68a6418c3317f8a4c78e)

Comment on file

~~~python
    def test_requires_packages_list(self) -> None:
        """The helper rejects metadata without a list-valued packages key."""
        with patch(
            "tests.workflows.workflow_test_helpers._cargo_metadata",
            return_value={"packages": "not-a-list"},
        ):
            with pytest.raises(
                AssertionError,
                match="cargo metadata must include a packages list",
            ):
                workflow_test_helpers._workspace_package_metadata("whitaker-installer")
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/71836/delta?repo-id=1205552&review-id=198&biomarker=Code+Duplication&filename=tests%2Fworkflows%2Ftest_workflow_test_helpers.py&method=)
The module contains 2 functions with similar structure: TestWorkspacePackageMetadata.test_reports_missing_workspace_package,TestWorkspacePackageMetadata.test_requires_packages_list

File: tests/workflows/test_workflow_test_helpers.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+class TestCargoMetadata:
+    """Tests for `_cargo_metadata()` error handling and edge cases."""
+
+    def test_requires_cargo_on_path(self) -> None:
+        """The helper fails fast when `cargo` cannot be resolved."""
+        with patch("tests.workflows.workflow_test_helpers.shutil.which", return_value=None):
+            with pytest.raises(
+                AssertionError,
+                match="cargo executable must be available in PATH",
+            ):
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/workflows/test_workflow_test_helpers.py]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: installer/tests/behaviour_docs.rs [hunk 2]
```diff
@@ -74,7 +74,7 @@ fn given_revision_pinning_metadata(toml_world: &TomlWorld) {
 
 #[given("a workspace metadata example with pre-built path")]
 fn given_prebuilt_path_metadata(toml_world: &TomlWorld) {
-    let block = find_block_containing("path = ");
+    let block = find_block_containing(r#"path = "/home/user/.local/share/whitaker/lints/"#);
```
[the remark this answers is not in the document]
Comment (leynos) [reply installer/tests/behaviour_docs.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

- The new tests patch `tests.workflows.workflow_test_helpers` by fully qualified import strings; consider patching attributes on the imported `workflow_test_helpers` module directly (e.g. `patch.object(workflow_test_helpers, "shutil")`) to make the tests more robust to module path refactors.
- In `behaviour_docs.rs`, the Given step is now coupled to a very specific prebuilt path string; if the docs example ever changes, this will become brittle—consider matching on a more stable substring (e.g. a suffix or directory fragment) while still keeping the test deterministic.
